app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt

```diff
@@ -188,6 +190,11 @@ class LaunchActivity : AppCompatActivity() {
                 }
                 ApiResponseStatus.REDIRECT -> apiResponse.uri?.let(::processInternalLink)
                 else -> {
+                    if (apiResponse.error?.exception is ApiController.NetworkException) {
+                        withContext(Dispatchers.Main) { showToast(R.string.errorNetwork) }
+                        finishAffinity()
+                        exitProcess(0)
```

Comment:
Why ?

Reply:
If the user has no network and can't load the share, we need to close the app because it failed to open.
The user can then try again to click on the link if he has a better connection